### PR TITLE
[core] clean up cudagraph batchsize padding logic

### PR DESCRIPTION
**Pull request by youkaichao**
unify the logic of cudagraph batchsize padding, and allow user-specified cudagraph capture sizes.

users might want to customize cudagraph capture sizes to accelerate start time, while only capture cudagraph for certain sizes they care about.

### CODE REVIEW

File: vllm/config.py [hunk 1]
```diff
@@ -2411,40 +2471,29 @@ class VllmConfig:
                                                  init=True)  # type: ignore
     instance_id: str = ""
 
-    @staticmethod
-    def get_graph_batch_size(batch_size: int) -> int:
-        """Returns the padded batch size given actual batch size.
-
-        Batch sizes are 1, 2, 4, _BATCH_SIZE_ALIGNMENT,
-        2*_BATCH_SIZE_ALIGNMENT, 3*_BATCH_SIZE_ALIGNMENT...
+    def model_pad_for_cudagraph(self, batch_size: int) -> int:
```
Comment (WoosukKwon) [inline vllm/config.py:2474]: nit:
```suggestion
    def pad_for_cudagraph(self, batch_size: int) -> int:
```

File: vllm/model_executor/models/jamba.py [hunk 2]
```diff
@@ -420,6 +420,9 @@ def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
 
         self.make_empty_intermediate_tensors = (
             self.model.make_empty_intermediate_tensors)
+        self.max_batch_size = (vllm_config.pad_for_cudagraph(
+            self.scheduler_config.max_num_seqs)
+                               if self.scheduler_config else 8192 + 2)
```
Comment (youkaichao) [inline vllm/model_executor/models/jamba.py:425]: @tlrmchlsmth do you know when `self.scheduler_config` will be `None` here?

File: vllm/config.py (hunk 1 above)
Comment (youkaichao) [reply vllm/config.py]: fixed now.

File: vllm/model_executor/models/jamba.py (hunk 2 above)
Comment (tlrmchlsmth) [reply vllm/model_executor/models/jamba.py:425]: No, I don't. @mzusman do you know?

File: vllm/config.py [hunk 3]
```diff
@@ -2491,40 +2514,10 @@ class VllmConfig:
...
-        _BATCH_SIZES_TO_CAPTURE.
-        """
-        padded_size = VllmConfig.get_graph_batch_size(max_num_seqs)
-        if padded_size in _BATCH_SIZES_TO_CAPTURE:
-            return padded_size
-        assert padded_size > _BATCH_SIZES_TO_CAPTURE[-1]
-        return _BATCH_SIZES_TO_CAPTURE[-1]
+    def pad_for_cudagraph(self, batch_size: int) -> int:
+        if batch_size > self.compilation_config.max_capture_size:
+            return self.compilation_config.max_capture_size
```
Comment (WoosukKwon) [inline vllm/config.py:2519]: Just wondering: should we raise an error (or assert False) in this case?

Comment (youkaichao) [reply vllm/config.py:2519]: I'm afraid some tests might reach this branch.

Comment (youkaichao) [reply vllm/config.py]: removed in [f4a7a77](https://github.com/vllm-project/vllm/pull/10996/commits/f4a7a77dabcbc132dd66e5c75fac0ed0244c0207) . let's see if any tests need to be updated.

**Review by WoosukKwon — APPROVED** [review]
Thanks for the fix!

File: vllm/model_executor/models/jamba.py (hunk 2 above)
Comment (mzusman) [reply vllm/model_executor/models/jamba.py]: Shouldn't be `None`, just added it as a safety guard -  It was added before the whole `vllm_config` was available inside the modeling file

Comment (youkaichao) [reply vllm/model_executor/models/jamba.py]: thanks! can you try to remove it then?

Comment (mzusman) [reply vllm/model_executor/models/jamba.py]: yeah, will open a PR shortly

**Pull request merged**